test/interpolating/encoding-params.test.js

```diff
@@ -0,0 +1,67 @@
+var test  = require('tape')
+  , rhumb = require('../../src/rhumb')
+
+test('Interpolating should handle encoding of the name for a non-empty variable part', function (t) {
+  t.plan(3)
+
+  t.equal(rhumb.interpolate('/wibble/{date%2Dof%2Dbirth}', { 'date-of-birth': 'bar' }), '/wibble/bar'
+    , 'returns path where variable substituted value when being interpolated')
+
+  t.equal(rhumb.interpolate('/wibble/{ns%2Fname}', { 'ns/name': 'bar' }), '/wibble/bar'
+    , 'returns path where variable substituted value when being interpolated')
+
+  t.equal(rhumb.interpolate('/wibble/{ns%252Fname}', { 'ns%2Fname': 'bar' }), '/wibble/bar'
+    , 'returns path where variable substituted value when being interpolated')
+})
```

Comment:
My motivation for doing this was that we decode query param names (keys) as well as the values.

My thoughts was to only allow names that matched `/[A-Za-z0-9%]+/`, so `date-of-birth` would not be valid).

If we keep this, then we need to make sure the docs explain this.  If this complicates things, then we can remove this.